docs/content/concepts/partitions-schedules-sensors/sensors.mdx

```diff
@@ -183,6 +183,62 @@ def asset_a_and_b_sensor_with_skip_reason(context):
+
+```python file=/concepts/partitions_schedules_sensors/sensors/asset_reconciliation_sensor.py startafter=start_asset_reconciliation_sensor endbefore=end_asset_reconciliation_sensor
+@repository
+def repository_1():
+    return [
+        asset_a,
+        asset_b,
+        asset_c,
+        build_asset_reconciliation_sensor(
+            asset_selection=AssetSelection.assets(asset_b, asset_c),
+            name="asset_reconciliation_sensor",
+        ),
+    ]
+```
+
+This sensor will materialize `asset_b` whenever any of its parents materialize, and it will materialize `asset_c` whenever any of its parents materialize.
+
+By default the created sensor will materialize an asset in the `asset_selection` whenever any of its parents materialize. You can modify this behavior to wait until all parents have materialized:
+
+```python file=/concepts/partitions_schedules_sensors/sensors/asset_reconciliation_sensor.py startafter=start_asset_reconciliation_sensor_wait_for_upstream endbefore=end_asset_reconciliation_sensor_wait_for_upstream
+@repository
+def repository_2():
+    return [
+        asset_a,
+        asset_b,
+        asset_c,
+        build_asset_reconciliation_sensor(
+            asset_selection=AssetSelection.assets(asset_b, asset_c),
+            name="asset_reconciliation_sensor",
+            wait_for_all_upstream=True,
+        ),
+    ]
+```
+
+To try to avoid unnecessary materializations of assets, the reconciliation sensor will wait to materialize an asset if one of its parents is currently materializing. You can override this so that the sensor will materialize the asset even if a parent is currently materializing:
+
+```python file=/concepts/partitions_schedules_sensors/sensors/asset_reconciliation_sensor.py startafter=start_asset_reconciliation_sensor_wait_for_in_progress endbefore=end_asset_reconciliation_sensor_wait_for_in_progress
+@repository
+def repository_3():
+    return [
+        asset_a,
+        asset_b,
+        asset_c,
+        build_asset_reconciliation_sensor(
+            asset_selection=AssetSelection.assets(asset_b, asset_c),
+            name="asset_reconciliation_sensor",
+            wait_for_in_progress_runs=False,
+        ),
+    ]
+```
+
+**Note:** If you have an asset whose parents update frequently, you should consider setting `wait_for_in_progress_runs=False` to avoid a deadlock where the reconciliation sensor cannot materialize an asset because each time the sensor evaluates, there is a parent asset materializing. 
```

Comment:
How does this explanation sound? I've taken a couple passes at making it clear, but i'm definitely struggling with it a bit. will continue to iterate if it's unclear

Reply:
Maybe it's worth mentioning the default behavior? I.e.:

By default, the asset reconciliation sensor does not materialize any asset whose parent is currently materializing. To avoid a deadlock for an asset whose parents update frequently, you can set `wait_for_in_progress_runs=False` to kick off a materialization regardless of whether any run in progress exists for a parent asset.

Reply:
yeah reiterating that would be good